### ISSUE TITLE
loss of data points at the very end of tests

### ISSUE THREAD

**Issue by jyoprakashJP** [johrstrom/jmeter-prometheus-plugin#58]
I tried putting 30 sec or more to the prometheus.delay property, but as soon as Jmeter test stop, metrics http server is down immediately and Prometheus scrapper loses the data. how to resolve this?

**Comment by chiabre** [comment]
I would suggest to add a empty/idle 'virtuale user'/thread at the end of
your test plan.

Please consider that the http server goes down as soon the test plan ends
since it is spawned within the jmeter java process

Luca

Il mer 21 ago 2019, 01:49 jyoprakashJP <notifications@github.com> ha
scritto:

**Comment by jyoprakashJP** [comment]
yes, I already tried with teardown thread group, and it works that way

**Comment by johrstrom** [comment]
Looking at the [Javadocs](https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpServer.html#stop-int-), it appears my understanding of delay is not what I had thought. It looks like the server will probably just stop immediately because it likely does not have any traffic.

Marking this as a bug because as a user, you've expected it to behave in a way that it does not, and I agree that that is the behaviour that should happen.

It seems like there's a workaround with is good, but, nonetheless, this feature seems to be broken.

**Comment by 10Zery** [comment]
Hello, were you able to fix the bug? If not, could you please suggest a workaround?

**Comment by johrstrom** [comment]
I feel like #116 should have fixed this. Can you try an updated version? Looks like that change made it into 0.6.1.
